kamu-adapter-http/src/smart_protocol/ws_tungstenite_client.rs

```diff
@@ -320,6 +322,110 @@ impl WsSmartTransferProtocolClient {
         Ok(dataset_objects_push_response)
     }
 
+    async fn export_group_of_object_files(
+        &self,
+        socket: &mut TungsteniteStream,
+        push_objects_response: DatasetPushObjectsTransferResponse,
+        src: Arc<dyn Dataset>,
+        transfer_options: ObjectTransferOptions,
+    ) -> Result<(), SyncError> {
+        let total_files_count = push_objects_response.object_transfer_strategies.len();
+        let uploaded_files_counter = Arc::new(AtomicI32::new(0));
+        let cloned_counters: Vec<_> = iter::repeat_with(|| uploaded_files_counter.clone())
+            .take(total_files_count)
+            .collect();
```

Comment:
Took me a while to understand what `cloned_counters` is...

Can this initialization and `zip` later on be replaced with something like:
```rust
let task_data: Vec<_> = push_objects_response.object_transfer_strategies
  .into_iter(|s| (s, uploaded_files_counter.clone())
  .collect();

futures::stream::iter(task_data)
  .try_for_each_concurrent(|s, c| ...)
  .await
```

Reply:
I agree, same sense, but slightly cleaner.